internal/resource/resource.go

```diff
@@ -145,6 +133,115 @@ func (r *Resource) patchSetsDeletionTimestamp() bool {
 	return dt != ""
 }
 
+type SchemaGetter interface {
+	Get(ctx context.Context, gvk schema.GroupVersionKind) (typeref *smdschema.TypeRef, schem *smdschema.Schema, err error)
```

Comment:
We pass a getter to Merge instead of just passing the values because it may need to invalidate the cache in the future in cases where the schema might be stale.